### PR TITLE
feat: use supplier on preview and vendor on pdf download

### PR DESCRIPTION
**Pull request by PatrickDEissler**

### CODE REVIEW

File: print_format/request_for_quotation.jinja [hunk 1]
```diff
@@ -8,12 +8,18 @@
     }
 </style>
 
-{% set company_address_name = frappe.db.get_value("Dynamic Link", filters={"link_doctype": "Company", "link_name": doc.company, "parenttype": "Address"}, fieldname="parent") %}
-{% set cmp_addr = frappe.get_doc("Address", company_address_name) %}
+{% set company_addresses = frappe.get_all("Dynamic Link", filters={"link_doctype": "Company", "link_name": doc.company, "parenttype": "Address"}, fields=["parent"]) %}
+{% set cmp_addr = frappe.get_doc("Address", company_addresses[0]['parent']) %}
```
Comment (barredterra) [inline print_format/request_for_quotation.jinja:12]: This part doesn't need to change - actually the previous approach was better and is the way that is used in the other formats as well.

File: print_format/request_for_quotation.jinja [hunk 2]
```diff
@@ -8,12 +8,18 @@
...
-{% set supplier_address_name = frappe.db.get_value("Dynamic Link", filters={"link_doctype": "Supplier", "link_name": doc.supplier, "parenttype": "Address"}, fieldname="parent") %}
-{% set pay_addr = frappe.get_doc("Address", supplier_address_name) %}
+{% if doc.vendor %}
+    {% set supplier_handle = doc.vendor %}
+{% else %}
+    {% set supplier_handle = doc.suppliers[0].supplier %}
+{% endif %}
+{% set supplier_doc = frappe.get_doc("Supplier", supplier_handle) %}
+{% set supplier_addresses = frappe.get_all("Dynamic Link", filters={"link_doctype": "Supplier", "link_name": supplier_handle, "parenttype": "Address"}, fields=["parent"]) %}
+{% set pay_addr = frappe.get_doc("Address", supplier_addresses[0]['parent']) %}
```
Comment (barredterra) [inline print_format/request_for_quotation.jinja:22]: This part doesn't need to change (apart from replacing `doc.supplier` with `supplier_doc.name`).

File: print_format/request_for_quotation.jinja [hunk 3]
```diff
@@ -33,7 +39,7 @@
             {%- endif %}
             </p>
         </div>
-        {{ doc.vendor }}<br />
+        {{ doc.vendor or supplier_doc.name }}<br />
```
Comment (barredterra) [inline print_format/request_for_quotation.jinja:42]:
```suggestion
        {{ supplier_doc.name }}<br />
```

When `doc.vendor` is available, `supplier_doc.name` will also be available and identical.

**Review by barredterra — CHANGES_REQUESTED** [review]

File: print_format/request_for_quotation.jinja (hunk 1 above)
Comment (PatrickDenis-stack) [reply print_format/request_for_quotation.jinja:12]:
```suggestion
{% set company_address_name = frappe.db.get_value("Dynamic Link", filters={"link_doctype": "Company", "link_name": doc.company, "parenttype": "Address"}, fieldname="parent") %}
{% set cmp_addr = frappe.get_doc("Address", company_address_name) %}
```
Alright, thanks. In this suggestion is the previous approach copied.

File: print_format/request_for_quotation.jinja (hunk 2 above)
Comment (PatrickDenis-stack) [reply print_format/request_for_quotation.jinja:22]:
```suggestion
{% set supplier_doc = frappe.get_doc("Supplier", supplier_handle) %}
{% set supplier_address_name = frappe.db.get_value("Dynamic Link", filters={"link_doctype": "Supplier", "link_name": supplier_doc.name, "parenttype": "Address"}, fieldname="parent") %}
{% set pay_addr = frappe.get_doc("Address", supplier_address_name) %}
```
Thanks. I tested it and it works. 
I just needed to additionally set the variable supplier_doc

**Pull request closed without merging**